sdk/python/kfp_tekton/compiler/compiler.py

```diff
@@ -388,6 +388,24 @@ def _create_pipeline_workflow(self, args, pipeline, op_transformers=None, pipeli
       if pipeline_conf.timeout:
         pipelinerun['spec']['timeout'] = '%ds' % pipeline_conf.timeout
 
+      # generate the Tekton service account template
+      service_template = {}
+      if len(pipeline_conf.image_pull_secrets) > 0:
+        service_template = {
+          'apiVersion': 'v1',
+          'kind': 'ServiceAccount',
+          'metadata': {'name': 'secrets'}
+        }
+        service_template['imagePullSecrets'] = []
+        image_pull_secrets = []
```

Comment:
should no need this line, same as service_template

Reply:
ok, thanks

Reply:
If I remove `image_pull_secrets = []`, can't directly use `image_pull_secrets` to do the append, it complains variable not defined. I removed service_template.